core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -62,6 +63,9 @@
       ImmutableSet.of(DataOperations.OVERWRITE, DataOperations.REPLACE, DataOperations.DELETE);
   private static final Set<String> VALIDATE_DATA_FILES_EXIST_SKIP_DELETE_OPERATIONS =
       ImmutableSet.of(DataOperations.OVERWRITE, DataOperations.REPLACE);
+  // delete files are only added in "overwrite" operations
+  private static final Set<String> VALIDATE_REPLACED_DATA_FILES_OPERATIONS =
```

Comment:
nit: I am not sure the var name conveys its purpose and does not follow the convention compared used in 2 vars above. If I understand correctly, this is a set of operations that may produce delete files.

Reply:
I'll update this. And I'll also add "delete" as an operation. Just because we only create "overwrite" commits for row-level deletes today doesn't mean that we won't create "delete" commits later. I considered updating the `RowDelta` operation so that it created delete commits if only delete files were added, but didn't end up doing it because we don't do that for overwrites that only remove data files.

We can discuss whether to convert overwrite commits to delete commits that later, but for now I think that we should add "delete" to this list because it would be a valid commit. Just because the reference implementation doesn't produce them doesn't mean it isn't valid.

Reply:
I had the same thought about using DELETE operation types when we just add delete files.

Reply:
Actually, I think the variable name does follow the convention. If we are validating replaced data files then those are the types of commits that are needed for validation.

I'm going to update this by adding `delete`, but not renaming.